### PR TITLE
definitions trimmed migration scripts

### PR DESCRIPTION
**Pull request by mpcen**

### CODE REVIEW

File: definitions-trimmed-migrate/src/services/migration-service/MigrationService.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                    // Handles a rare edge-case where a blob was deleted
+                    // after it was queued by the blob-iterator
+                    if (!definitionBlob) {
+                        const error = new Error(`Blob ${blobName} no longer exists`);
+
+                        this.logService.error('RedisService.get', error, {
+                            exception: error,
+                        });
+
+                        await this.markProcessed(blobName, message);
```
Comment (qtomlinson) [inline definitions-trimmed-migrate/src/services/migration-service/MigrationService.ts:73]: shall we increment the blob counter here (similar to line 88 to 91)?

File: definitions-trimmed-migrate/src/services/migration-service/MigrationService.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    private async downloadBlob(blobName: string): Promise<DefinitionBlob> {
+        return await this.azureStorageService.downloadBlob(blobName);
+    }
+
+    private async getQueuedMessages(batchSize: string) {
+        return await this.azureStorageService.receiveMessages(Number(batchSize));
+    }
+
+    private async getDefinitionById(_id: string) {
+        const document = await this.mongoService.findOne(this.options.mongoOptions.collectionName, {
```
Comment (qtomlinson) [inline definitions-trimmed-migrate/src/services/migration-service/MigrationService.ts]: nit: return await this.mongoService.findOne

File: definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        const startTime = new Date();
+        let i = 1;
+        let queued = 0;
+
+        try {
+            for await (const blob of this.containerClient.listBlobsFlat()) {
+                const { name, properties } = blob;
+                const { lastModified } = properties;
+
+                if (lastModified < startTime && !(await this.redisService.get(name))) {
```
Comment (qtomlinson) [inline definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts]: check for not existent instead? processed entries with value 0 in the redis cache will pass this check.

File: definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                exception: err as Error,
+            });
+
+            throw err;
+        }
+    }
+
+    public async receiveMessages(count: number) {
+        try {
+            return await this.queueClient.receiveMessages({ numberOfMessages: count });
```
Comment (qtomlinson) [inline definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts:67]: Is the default visibility timeout long enough for our download and subsequent processing?

**Review by qtomlinson — APPROVED** [review]

**Review by qtomlinson — APPROVED** [review]
The diagram is very helpful in understanding the migration.  A few minor adjustment: 3, stores a trimmed version ... when it is not present.  at 4. blob processor not longer deletes the cache entry.

File: definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts (hunk 4 above)
Comment (mpcen) [reply definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts:67]: changed to 900 seconds (15 min) to be safe

**Pull request merged**